include/cell.tcc

```diff
@@ -856,6 +856,58 @@ Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate(
   return strain_rate;
 }
 
+//! Compute strain rate for reduced integration at the centroid of cell
+template <unsigned Tdim>
+Eigen::VectorXd mpm::Cell<Tdim>::compute_strain_rate_centroid(unsigned phase) {
+  // Define strain rate
+  Eigen::VectorXd strain_rate;
+
+  switch (Tdim) {
+    case (1): {
+      strain_rate.resize(1);
+      break;
+    }
+    case (2): {
+      strain_rate.resize(3);
+      break;
+    }
+    default: {
+      strain_rate.resize(6);
+      break;
+    }
+  }
+
+  strain_rate.setZero();
+
+  // Get centroid local coordinates, which are zeros
+  Eigen::Matrix<double, Tdim, 1> xi_centroid;
+  xi_centroid.setZero();
+
+  // Initialize B-Matrix
+  std::vector<Eigen::MatrixXd> bmatrix;
+
+  try {
+    // Get B-Matrix at the centroid
+    bmatrix = element_->bmatrix(xi_centroid);
+    for (unsigned i = 0; i < bmatrix.size(); ++i)
+
+      // Check if B-Matrix size and number of nodes match
```

Comment:
Remove this check, this does not apply to this function

Reply:
I think it still does - the matrix size should be correct before we proceed in calculating
Note, it's like talking about a particle in a cell

Reply:
You are getting the B-matrix for the element 2 lines above, so this check doesn't apply.

Reply:
Understood